piksi_tools/console/settings_list.py

```diff
@@ -9,36 +32,55 @@ def __str__(self):
     pp.pprint(self.list_of_dicts)
 
   def get_dict(self, group, name):
-    for element in self.list_of_dicts:
-        if element['name'] == name and element['group'] == group:
-          return element
+    #confirm our class actually has list_of_dicts
+    if len(self.list_of_dicts) != 0 :
+      for element in self.list_of_dicts:
+          if element['name'] == name and element['group'] == group:
+            return element
 
   def get_field(self, group, name,field):
-    thisdict = self.get_dict(group, name)
     returnvar = ""
-    if thisdict and isinstance(thisdict, dict):
-      returnvar = thisdict.get(field,"")
-    else:
-      print "Error in settings list parsed yaml file."
-      print "No entry for name  {0} and group is {1}".format(name,group)
+    #confirm our class actually has list_of_dicts
+    if len(self.list_of_dicts) != 0 :
+      thisdict = self.get_dict(group, name)
+      if thisdict and isinstance(thisdict, dict):
+        returnvar = thisdict.get(field,"")
+      else:
+        print "Error in settings list parsed yaml file."
+        print "No entry for name  {0} and group is {1}".format(name,group)
     if not returnvar:
       returnvar = ""
     return returnvar
 
   def return_groups(self):
-    group_set = set()
     output = []
-    for element in self.list_of_dicts:
-        group = element['group']
-        if group not in group_set:
-          output.append(element['group'])
-          group_set.add(group)
+    #confirm our class actually has list_of_dicts
+    if len(self.list_of_dicts) != 0 :
```

Comment:
Instead of always checking if the length of the dicts is not zero, how about initializing `list_of_dicts` to `{}`, and then updating it if you're able to load the yaml successfully.